File: app/src/main/java/com/projectcklmp/a2805_1308_map/VideoPlayer.java
package com.projectcklmp.a2805_1308_map;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.VideoView;
import com.projectcklmp.a2805_1308_map.R;

public class VideoPlayer extends Activity {
    private VideoView mVideoView;



    @Override
    protected void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        String videoUrl = getIntent().getStringExtra("videoUri");
        System.out.println("Hello" + videoUrl);
        setContentView(R.layout.content_video_player);
        mVideoView = findViewById(R.id.VideoView);
        Uri videoUri =Uri.parse(videoUrl);
        mVideoView.setVideoURI(videoUri);
        mVideoView.start();
    }
}

File: app/src/main/java/com/projectcklmp/a2805_1308_map/VideoPlayer.kt
package com.projectcklmp.a2805_1308_map

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.VideoView

import android.media.MediaPlayer

import android.support.v4.content.ContextCompat


class VideoPlayer : Activity() {
    private var mVideoView: VideoView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val videoUrl = intent.getStringExtra("videoUri")
        setContentView(R.layout.content_video_player)
        mVideoView = findViewById(R.id.VideoView)
        val videoUri = Uri.parse(videoUrl)
        mVideoView!!.setVideoURI(videoUri)
        mVideoView!!.start()



        mVideoView!!.setOnCompletionListener {
            finish()
        }
    }


}